src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -47,25 +58,38 @@ public void start(Map<String, String> settings) {
         } catch (ChannelSinkConnectorConfig.ConfigException e) {
             logger.error("Failed to create Ably client", e);
         }
-        ablyClient.connect(isSuspended -> {
-            synchronized(ChannelSinkTask.this){
-                suspended.set(isSuspended);
-                if (!isSuspended){
-                    processSuspendQueue();
-                }
-            }
-        });
+//        ablyClient.connect(isSuspended -> {
+//            synchronized(ChannelSinkTask.this){
+//                suspended.set(isSuspended);
+//                if (!isSuspended){
+//                    processSuspendQueue();
+//                }
+//            }
+//        });
+        logger.info("Starting Ably channel Sink task");
+        // start the Batch processing thread.
+        this.sinkRecords = new ConcurrentLinkedQueue<>();
+        this.batchProcessingThread = new BatchProcessingThread(this.sinkRecords, this.ablyClient);
```

Comment:
Does this not mean that the entire queue of incoming records are being handed to a single worker thread? How do we break off chunks to be sent in each batch request?

Reply:
the logic is create threads based on the thread pool size. And the worker threads would poll elements from the queue and call the Batch API.